Assets/Scripts/PlayerSelect/PlayerSelectManager.cs

```diff
@@ -0,0 +1,159 @@
+    /// </summary>
+    /// <param name="player"></param>
+    /// <param name="playerName"></param>
+    /// <param name="playerID"></param>
+    public void setPlayerNameTag(GameObject player, string playerName, int playerID)
+    {
+        // Check if nameTag exists already
+        bool nameExists = false;
+        for (int i = 0; i < playerNames.Count; i++)
+        {
+            if (playerNames[i].text == playerName)
+            {
+                nameExists = true;
+            }
+        }
+
+        // If it doesn't, create new nametag based on player name
+        if (!nameExists)
+        {
+            Vector3 playerPosition = new Vector3(player.transform.position.x, 2, player.transform.position.z);
+
+            TMP_Text name = nameTags[playerID];
+            name.GetComponent<Transform>().position = playerPosition; // Sets the nametag position over the head of the player model
+            name.text = playerName; // Sets the text of the nametag
+            playerNames.Add(name); // Adds TMP_text object to list for monitoring
+        }
+    }
+
+    /// <summary>
+    /// Starts or stops the playerselect animations coroutine. Called by "Start" and "Back" buttons in menu.
+    /// </summary>
+    /// <param name="play"></param>
+    public void controlAnimations(bool play)
+    {
+        if(play)
+        {
+            // Start random animations coroutine for playerselect
+            StartCoroutine("PlayRandomAnimation");
+        }
+        else
+        {
+            // Stop random animations coroutine for playerselect
+            StopCoroutine("PlayRandomAnimation");
+        }
+    }
+
+    /// <summary>
+    /// Coroutine that continuously plays random idle animations in the player select menu
+    /// </summary>
+    IEnumerator PlayRandomAnimation()
+    {
+        yield return new WaitForSeconds(5f);
+        while (true)
+        {
+            int randomAnimatorNumber = Random.Range(0, playerInputManagerController.playerInputs.Count); // Choose random playermodel to animate
+
+            Animator randomAnimator = playerAnimators[randomAnimatorNumber]; // Get the animator for one of the players that has a connected input
+
+            // If randomAnimatorNumber is player all the way to the right, don't include cardpeek trigger
+            string randomTrigger = "";
+            if (randomAnimatorNumber == playerInputManagerController.playerInputs.Count - 1)
+            {
+                randomTrigger = randomAnimator.GetParameter(Random.Range(2, randomAnimator.parameterCount)).name; // Choose a random trigger to set, excluding CardPeek
+            }
+            else if(randomAnimatorNumber == 0)
+            {
+                while(randomTrigger != "CardPeekReaction")
+                {
+                    randomTrigger = randomAnimator.GetParameter(Random.Range(0, randomAnimator.parameterCount)).name; // Choose a random trigger to set, excluding CardPeekReaction
```

Comment:
Can't you store the parameter names upon creation and then filter out the unwanted reaction name here? Using a `while` with a random condition is a little scary, though the probability for a long time with one outcome is miniscule.

And with a list of names, you would be able to use the `RandomElement()` extension method we've prepared for this kinda thing.

...oh, and then I noticed that you *know* that CardPeekReaction is the last parameter already (from line 145), perhaps you could just replicate that line here instead?

Reply:
Redolved using a list of parameternames and RandomElement().